Fix/update extending tina docs (#1293)
tinacms/tinacms.org: content/docs/extending-tina/overview.md

---
title: Extending Tina
id: '/docs/extending-tina/overview'
prev: null
next: /docs/extending-tina/validation
---

Tina has many advanced features that allow the entire CMS editing experience to be customized.

## Customizing Fields

> Tina allows the forms to be customized thought the `ui` property. This makes experience custom for your editors. Some of the main customization features are; 

Reviewer: ```suggestion
Tina allows the forms to be customized through the `ui` property. This allows a custom experience to be created for your editors. Some of the main customization features are:
```

> - Adding [frontend validation](/docs/extending-tina/validation/) to a component so that it must be saved with the desired format

Reviewer: ```suggestion
- Adding [client-side validation](/docs/extending-tina/validation/) to a component so that a desired format can be enforced before saving
```

> - Customizing [the field component](/docs/extending-tina/custom-field-components/) that is rendered

Reviewer: ```suggestion
- Customizing the rendered [field component](/docs/extending-tina/custom-field-components/)
```

> - Customizing how the users input is [parsed and formatted](/docs/extending-tina/format-and-parse/)

Reviewer: ```suggestion
- Customizing the [parsing and formatting](/docs/extending-tina/format-and-parse/) of user input
```

## Customizing the CMS instance

The `.tina/schema.ts` config has an optional `cmsCallback` parameter that can be added to customize the CMS instance.

```diff
// .tina/schema.ts

// ...
export config = defineConfig({
  apiURL,
+ cmsCallback: (cms) => {
+   cms.sidebar.position = 'overlay'
+   return cms
+ }
})
```

> The `cmsCallback` hook might be used to alter Tina's UI, dynamically hide the sidebar on certain pages, tap into the CMS event bus, etc, but the most common use-case is for registering custom field plugins.

Reviewer: ```suggestion
The `cmsCallback` hook is primarily used for registering custom field plugins. It can also be used for altering Tina's UI, dynamically hiding the sidebar on specific pages, tapping into the CMS event bus, etc.
```
